File: Python3/728. Self Dividing Numbers.py
class Solution:
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        numlist = []
        for num in range(left, right+1):
            request = True
            for digit in str(num):
                if digit == '0' or num%int(digit):
                    request = False
            if request:
                numlist.append(num)
        return numlist
        # 精彩至极的python一行列表推导式
        # return [x for x in range(left, right+1) if all(y and not x%y for y in map(int,str(x)))]

File: Golang/728. Self Dividing Numbers.go
func selfDividingNumbers(left int, right int) []int {
    var result []int
    var i int
    for i = left; i <= right; i++ {
        var num int = i
        request := true
        res := 0
        for {
            res = num%10
            if res == 0 || i%res != 0 {
                request = false
                break
            }
            num = num/10
            if num == 0 {
                break
            }
        }
        if request {
            result = append(result, i)
        }
    }
    return result
}